src/java/main/org/apache/zookeeper/server/NIOServerCnxnFactory.java

```diff
@@ -275,20 +307,9 @@ public synchronized void closeSession(long sessionId) {
 
     @SuppressWarnings("unchecked")
     private void closeSessionWithoutWakeup(long sessionId) {
-        HashSet<NIOServerCnxn> cnxns;
-        synchronized (this.cnxns) {
-            cnxns = (HashSet<NIOServerCnxn>)this.cnxns.clone();
-        }
-
-        for (NIOServerCnxn cnxn : cnxns) {
-            if (cnxn.getSessionId() == sessionId) {
-                try {
-                    cnxn.close();
-                } catch (Exception e) {
-                    LOG.warn("exception during session close", e);
-                }
-                break;
-            }
+        NIOServerCnxn cnxn = sessionMap.remove(sessionId);
+        if (cnxn != null) {
+            cnxn.close();
```

Comment:
Why did you remove the `try-catch` block around `cnxn.close()`? We still can have exceptions being thrown during `cnxn.close()`, right?